devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -517,6 +529,9 @@ private void handleAutoCompile() throws MojoExecutionException {
         if (compileNeeded) {
             triggerCompile(false, prepareNeeded);
         }
+        if (indexClassNeeded) {
+            initClassIndexes();
```

Comment:
I moved it up here to have a more logical order (`main` stuff before `test` stuff).